### ISSUE TITLE
OrgChart: FontAwesome icons no longer work

### ISSUE THREAD

**Issue by melloware** [primefaces-extensions/primefaces-extensions#1050]
@jxmai your recent changes prevent FontAwesome icons from working in the OrgChart.

In our showcase `parentNodeSymbol="fa-folder-open"` now does nothing.

**Comment by melloware** [comment]
Looks like the Taglib is also wrong now "Using font awesome icon to imply that the node has child nodes. Default is "fa-users" "  It looks like Org Chart sort of half baked their own icons which are not really icons?

**Comment by jxmai** [comment]
I can review further.

the most recent change regarding orgchart was https://github.com/primefaces-extensions/primefaces-extensions/pull/1018 with running screenshot

and it should be version 12.0.3

the current showcase is using 12.0.1, which does not have the changes

https://www.primefaces.org/showcase-ext/sections/orgchart/basicUsage.jsf

![image](https://user-images.githubusercontent.com/10467831/216505064-ecdb4a74-8444-419b-b0aa-41ae0b96fbf8.png)

They look identical and I can see the folder open icon. Is there anything I missed there? Please let me know

**Comment by melloware** [comment]
Here is what I see with 12.0.3

![image](https://user-images.githubusercontent.com/4399574/216597736-138bac12-dc8a-4244-b994-66ed40d0e44c.png)

And if you look at the generated HTML they are now using `oci` instead of `fa` like in previous versions.

![image](https://user-images.githubusercontent.com/4399574/216597823-24c58411-90da-459d-b2c8-95c4c6b739b2.png)

It seems like they tried to make their own icons and now fontawesome does not work.  

here is the HTML in 12.0.1

![image](https://user-images.githubusercontent.com/4399574/216598121-3bc87d8d-e47e-4b7e-b47b-25b19b4ee3fc.png)

**Comment by jxmai** [comment]
I see. Let me self-assign this ticket and I will review this one.

From their website,

https://dabeng.github.io/OrgChart/default-collapsed.html

I can see the newly introduced oci

![image](https://user-images.githubusercontent.com/10467831/216621222-3a5eb49c-0323-4d50-a11d-a078d70dcd2a.png)

based on the comments, it says custom icons indeed

![image](https://user-images.githubusercontent.com/10467831/216621483-1077c249-e36a-45c0-a432-73cf0d8e8dc5.png)

and there's some commits in orgchart repo to remove the font-awesome dependency

https://github.com/dabeng/OrgChart/commit/f44deb8d801bbb58f2f024c456330e805814f0d9

and refactor the demo:

https://github.com/dabeng/OrgChart/commit/571fc97269493861867efe6b3ef3edf7f1fd8cc2

My initial impression is that font-awesome dependency is removed and they introduce their own icon indeed, but the question is that why it is not showing up in our demo that needs to be investigated.

**Comment by melloware** [comment]
It also looks like they have a new version 3.4.0 out.  It might be we need to somehow override their font with font-awesome hopefully they have some guidance on how to override their icons with custom icons?

**Comment by jxmai** [comment]
Yes, for 3.4.0. I gave it a try previously but it did not go too well (also mentioned in https://github.com/primefaces-extensions/primefaces-extensions/issues/1017) so I wanted to bring it up to the latest v2 before moving on v3.

If we upgrade v3 directly as of now, we will see the orgchart style is corrupted. I have to check what breaks v3 orgchart in their latest code in our integration.

If we can resolve this icon issue in v2, that would definitely one step closer for moving to v3 since v3 uses custom icons as well combining with other changes.

**Comment by melloware** [comment]
Agreed

**Comment by jxmai** [comment]
Note: The oci icon (orgchart customized icon is working if invoke correctly)

For example,

```css
oci oci-leader symbol
```

https://dabeng.github.io/OrgChart/default-collapsed.html

local showcase by changing the css directly via chrome web tool: 

12.0.4-SNAPSHOT

![image](https://user-images.githubusercontent.com/10467831/216731008-811325ab-12bf-4db3-8669-e2a3421c2e77.png)

which means I need to find the alternative in orgchart repo that can replace the previous font-awesome icon and update our code.

**Comment by melloware** [comment]
Yep we also need to update the taglib.xml for that property which says "fa-users" is the default because that is what it used to be.

**Comment by jxmai** [comment]
Sounds great. I will check further

**Comment by jxmai** [comment]
It looks to me that orgchart  removed the ability to define font awesome icon.

Currently the only icon left is oci-leader, so let's hope the icon support would increase in the future versions.

https://github.com/dabeng/OrgChart/blob/e8093ef5f1e1f1089e71df775d30277388e00700/dist/css/jquery.orgchart.css#L605-L627.

From PFE perspective, we will make two changes:

1. update the showcase to use parentNodeSymbol: `oci-leader` as https://github.com/dabeng/OrgChart/blob/e8093ef5f1e1f1089e71df775d30277388e00700/dist/js/jquery.orgchart.js#L33

2. Update taglib description to reflect the changes. The default was done behind the scene in orgchart.js so no change to the default value in JSF taglib other than the description.

I will provide the PR shortly.

Local preview: 

![image](https://user-images.githubusercontent.com/10467831/216740335-a0cc58db-1987-4538-a57c-a3dd00fbfc74.png)

**Comment by melloware** [comment]
I agree with those changes for now. 

I wonder if we should open a ticket on jquery orgchart. I think they should allow custom icons like PrimeFaces does.  It supports multiple different icon sets.

**Comment by jxmai** [comment]
Yes, I just open a ticket on jquery orgchart repo. https://github.com/dabeng/OrgChart/issues/695

**Comment by melloware** [comment]
See for solution to enable fontAwesome icons: https://github.com/dabeng/OrgChart/issues/695

**Comment by jxmai** [comment]
To leverage on the feature, we have to upgrade orgchart to version 3.7.0. I will assign a ticket to myself and will check when I have some time.
